crates/host/src/wasmbus/mod.rs

```diff
@@ -265,71 +219,72 @@ fn invocation_hash(
-    /// of preventing them from being forged over the network when connected to a lattice,
-    /// so an invocation requires a reference to the host (signing) key
-    pub fn new(
-        // package -> target -> entity
-        links: &HashMap<String, HashMap<String, WasmCloudEntity>>,
-        hostkey: &KeyPair,
-        origin: WasmCloudEntity,
-        target: Option<&TargetEntity>,
-        operation: impl Into<String>,
-        msg: Vec<u8>,
-    ) -> anyhow::Result<Invocation> {
-        const DEFAULT_LINK_NAME: &str = "default";
-
-        let operation = operation.into();
-        let (package, operation) = operation
-            .split_once('/')
-            .context("failed to parse operation")?;
-        let target = match target {
-            None => links
-                .get(package)
-                .and_then(|targets| targets.get(DEFAULT_LINK_NAME))
-                .context("link not found")?
-                .clone(),
-            Some(TargetEntity::Link(link_name)) => links
-                .get(package)
-                .and_then(|targets| targets.get(link_name.as_deref().unwrap_or(DEFAULT_LINK_NAME)))
-                .context("link not found")?
-                .clone(),
-            Some(TargetEntity::Actor(ActorIdentifier::Key(key))) => WasmCloudEntity {
-                public_key: key.public_key(),
-                ..Default::default()
-            },
-            Some(TargetEntity::Actor(ActorIdentifier::Alias(alias))) => {
-                // TODO: Support actor call aliases
-                // https://github.com/wasmCloud/wasmCloud/issues/505
-                bail!("actor-to-actor calls to alias `{alias}` not supported yet")
-            }
-        };
-        // TODO: Support per-interface links
-        let id = Uuid::from_u128(Ulid::new().into()).to_string();
-        let host_id = hostkey.public_key();
-        let target_url = format!("{}/{operation}", target.url());
-        let claims = jwt::Claims::<jwt::Invocation>::new(
-            host_id.to_string(),
-            id.to_string(),
-            &target_url,
-            &origin.url(),
-            &invocation_hash(&target_url, origin.url(), operation, &msg),
-        );
-        let encoded_claims = claims.encode(hostkey).context("failed to encode claims")?;
-
-        let operation = operation.to_string();
-        Ok(Invocation {
-            content_length: Some(msg.len() as _),
-            origin,
-            target,
-            operation,
-            msg,
-            id,
-            encoded_claims,
-            host_id,
-        })
-    }
+/// Creates a new invocation. All invocations are signed with the host key as a way
+/// of preventing them from being forged over the network when connected to a lattice,
+/// so an invocation requires a reference to the host (signing) key
+#[allow(clippy::implicit_hasher)]
+#[allow(clippy::missing_errors_doc)]
+pub fn new_invocation(
```

Comment:
Why the change from `Invocation::new`?

Reply:
There were types referenced in `new` that weren't easily moved into `wasmcloud_core`. I refactored the first commit to account for this, so it's back to `Invocation::new`